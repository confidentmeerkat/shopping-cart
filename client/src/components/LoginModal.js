import React from 'react';
import Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import '../styles/LoginModal.css';

const LoginModal = (props) => (
  <Modal
    className="login-modal"
    isOpen={props.isOpen}
    onRequestClose={props.onRequestClose}
  >
    <form className="form" action="/auth/login" method="POST">
      <h1>Log In</h1>
      <TextField
        hintText="Enter Username"
        floatingLabelText="Username"
        name="username"
        autoFocus
      /><br />
      <TextField
        hintText="Enter Password"
        floatingLabelText="Password"
        name="password"
        type="password"
      /><br />
      <RaisedButton 
        className="btn" 
        label="Submit" 
        primary={true} 
        type="submit"
      />
      <p>Don't have an account yet? <a onClick={props.switch}>Register here</a>.</p>
    </form>
  </Modal>
);

export default LoginModal;