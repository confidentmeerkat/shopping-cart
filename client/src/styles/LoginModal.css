.login-modal {
  width: 385px;
  height: 400px;
  margin: auto;
  margin-top: 100px;
  background: white;
  color: #2e3a4e;
  border: 1px solid white;
  border-radius: 5px;
  outline: none;
}

.login-modal .form {
  width: 70%;
  margin: auto;
  margin-top: 10px;
  text-align: center;
  font-family: 'Roboto';
}

.login-modal .form h1 {
  color: #00BCD4;
}

.login-modal .form .btn {
  margin: 30px auto 50px auto;
}

.login-modal .form p {
  margin: 0;
  font-style: italic;
}

.login-modal .form a {
  color: #00BCD4;
  cursor: pointer;
}