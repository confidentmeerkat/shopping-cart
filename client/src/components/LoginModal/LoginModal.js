import React from 'react';
import Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import '@styles/LoginModal.css';

const LoginModal = ({ isOpen, onRequestClose, toggle }) => (
  <Modal
    className="login-modal"
    isOpen={isOpen}
    onRequestClose={onRequestClose}
  >
    <form className="form" action="/auth/login" method="POST">
      <h1>Log In</h1>
      <TextField
        hintText="Enter Username"
        floatingLabelText="Username"
        name="username"
        autoFocus
      /><br />
      <TextField
        hintText="Enter Password"
        floatingLabelText="Password"
        name="password"
        type="password"
      /><br />
      <RaisedButton 
        className="btn"
        label="Submit"
        primary={true}
        type="submit"
      />
      <p>Don't have an account yet? <a onClick={toggle}>Register here</a>.</p>
    </form>
  </Modal>
);

export default LoginModal;
