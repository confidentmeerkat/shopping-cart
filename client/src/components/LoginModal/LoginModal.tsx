import * as React from 'react';
import * as Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import { ModalProps } from '@typings/modal';
import '@styles/LoginModal.css';

const LoginModal: React.SFC<ModalProps> = ({ isOpen, onRequestClose, toggle }): JSX.Element => (
  <Modal
    className="login-modal"
    isOpen={isOpen}
    onRequestClose={onRequestClose}
  >
    <form className="form" action="/auth/login" method="POST">
      <h1>Log In</h1>
      <TextField
        hintText="Enter Username"
        floatingLabelText="Username"
        name="username"
        autoFocus
      /><br />
      <TextField
        hintText="Enter Password"
        floatingLabelText="Password"
        name="password"
        type="password"
      /><br />
      <RaisedButton
        className="btn"
        label="Submit"
        primary={true}
        type="submit"
      />
      <p>Don't have an account yet? <a onClick={() => toggle}>Register here</a>.</p>
    </form>
  </Modal>
);

export default LoginModal;
