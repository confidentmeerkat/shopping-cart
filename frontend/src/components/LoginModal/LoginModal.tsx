import React, { useState } from 'react';
import Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import { ModalProps } from '@typings/modal';
import { login } from '@api/auth';
import '@styles/LoginModal.css';

const LoginModal = ({ isOpen, onRequestClose, setActiveModal, setUser }: ModalProps): JSX.Element => {
  const [data, setData] = useState({});

  const setFormField = (key: string, value: any) => {
    setData({
      ...data,
      [key]: value,
    });
  }

  const onSubmit = () => {
    login(data).then((res) => {
      setUser && setUser(res.data);
      onRequestClose();
    });
  }

  return (
    <Modal
      className="login-modal"
      isOpen={isOpen}
      onRequestClose={onRequestClose}
    >
      <div className="form">
        <h1>Log In</h1>
        <TextField
          hintText="Enter Username"
          floatingLabelText="Username"
          autoFocus
          onChange={({ target }: any) => setFormField('username', target.value)}
        /><br />
        <TextField
          hintText="Enter Password"
          floatingLabelText="Password"
          type="password"
          onChange={({ target }: any) => setFormField('password', target.value)}
        /><br />
        <RaisedButton
          className="btn"
          label="Submit"
          primary={true}
          onClick={onSubmit}
        />
        <p>Don't have an account yet? <a onClick={() => setActiveModal('register')}>Register here</a>.</p>
      </div>
    </Modal>
  );
}

export default LoginModal;
