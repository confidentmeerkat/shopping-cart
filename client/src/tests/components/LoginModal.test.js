import React from 'react';
import { shallow } from 'enzyme';
import LoginModal from '../../components/LoginModal';

describe('<LoginModal />', () => {
  it('renders LoginModal correctly', () => {
    const wrapper = shallow(<LoginModal />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').children()).toHaveLength(7);
  });
});