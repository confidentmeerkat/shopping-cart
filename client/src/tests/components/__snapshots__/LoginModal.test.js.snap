// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginModal /> renders LoginModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="login-modal"
  closeTimeoutMS={0}
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <form
    action="/auth/login"
    className="form"
    method="POST"
  >
    <h1>
      Log In
    </h1>
    <TextField
      autoFocus={true}
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Username"
      fullWidth={false}
      hintText="Enter Username"
      multiLine={false}
      name="username"
      rows={1}
      type="text"
      underlineShow={true}
    />
    <br />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Password"
      fullWidth={false}
      hintText="Enter Password"
      multiLine={false}
      name="password"
      rows={1}
      type="password"
      underlineShow={true}
    />
    <br />
    <RaisedButton
      className="btn"
      disabled={false}
      fullWidth={false}
      label="Submit"
      labelPosition="after"
      primary={true}
      secondary={false}
      type="submit"
    />
    <p>
      Don't have an account yet? 
      <a>
        Register here
      </a>
      .
    </p>
  </form>
</Modal>
`;
